assets/js/util/index.js

```diff
@@ -332,6 +332,45 @@ export const refreshAuthentication = async () => {
 	}
 };
 
+/**
+ * Gets data for all modules.
+ *
+ * Because googlesitekit.modules contains both module information (legacy) and
+ * API functions (new), we should be using this function and never access
+ * googlesitekit.modules directly to access module data.
+ *
+ * This function should be removed once this object is no longer used to store
+ * legacy module data.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}  _googlesitekit Optional. googlesitekit global; can be replaced for testing.
+ * @return {Object} Object with module data, with each module keyed by its slug.
+ */
+export const getModulesData = ( _googlesitekit = global.googlesitekit ) => {
```

Comment:
We can write this a bit more succinctly and without having to create the extra object:

```js
export const getModulesData = ( _googlesitekit = global.googlesitekit ) => {
    const modulesObject = _googlesitekit.modules;

	return Object.keys( modulesObject || {} ).reduce( ( acc, slug ) => {
		if ( 'object' !== typeof modulesObject[ slug ] ) {
			return acc;
		}

		if (
			'undefined' === typeof modulesObject[ slug ].slug ||
			'undefined' === typeof modulesObject[ slug ].name ||
			modulesObject[ slug ].slug !== slug
		) {
			return acc;
		}

        return { ...acc, slug: modulesObject[slug] };
	}, {} );
};
```

I personally like the `reduce()` rather than the `forEach` that modifies another object.